synapse/lib/ast.py

```diff
@@ -743,8 +743,8 @@ async def run(self, runt, genr):
                 if isinstance(name, (list, tuple)):
 
                     if len(name) != len(item):
-                        mesg = 'Number of items to unpack does not match the number of variables.'
-                        raise s_exc.StormVarListError(mesg=mesg, names=name, vals=item)
+                        mesg = f'Number of items to unpack does not match the number of variables: {repr(item)[:256]}'
+                        raise s_exc.StormVarListError(mesg=mesg, names=name, numitems=len(item))
```

Comment:
This could also blow up if `item` isn't an type iterable, like:
```ERROR: ('TypeError', {'efile': 'ast.py', 'eline': 745, 'mesg': "object of type 'int' has no len()"})```

Reply:
This is inside an isinstance check for list,tuple though

Reply:
That is checking that the name list is a list/tuple, not the item, so something like
```for ($x, $y) in ((1),) { $lib.print($x) }```
would still get here

Reply:
Updated; decided to let a Prim StormRuntimeError raise directly